### PR TITLE
 Add Lighthouse Module

### PR DESCRIPTION
**Pull request by Fred-sun**
##### SUMMARY

 Add Lighthouse Module.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

azure_rm_registrationdefinition.py
azure_rm_registrationdefinition_info.py
azure_rm_registrationassignment.py
azure_rm_registrationassignment_info.py
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/azure_rm_registrationassignment.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+---
+module: azure_rm_registrationassignment
+version_added: '1.3.0'
+short_description: Manage Azure RegistrationAssignment instance
+description:
+    - Create and delete instance of Azure RegistrationAssignment.
+options:
+    scope:
+        description:
+            - Scope of the registration assignment..
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_registrationassignment.py]: - Scope of the registration assignment. Can be in subscription or group level.

File: plugins/modules/azure_rm_registrationassignment.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+        required: true
+        type: str
+    registration_assignment_id:
+        description:
+            - ID of the registration assignment.
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_registrationassignment.py:31]: +: If is not specified, an UUID will be generated for it.

File: plugins/modules/azure_rm_registrationassignment.py [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+        scope: subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        registration_assignment_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        properties:
+          registration_definition_id: /subscriptions/xxx-xxx/providers/Microsoft.ManagedServices/registrationDefinitions/xxx-xxx
+
+
+    - name: Create Registration Assignment in resourcegroup level
+      azure_rm_registrationassignment:
+        scope: subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup
+        registration_assignment_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_registrationassignment.py]: remove this line, to indicate assignment id is not mandatory.

File: plugins/modules/azure_rm_registrationassignment.py [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+
+    - name: Create Registration Assignment in subscription level
+      azure_rm_registrationassignment:
+        scope: subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        registration_assignment_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        properties:
+          registration_definition_id: /subscriptions/xxx-xxx/providers/Microsoft.ManagedServices/registrationDefinitions/xxx-xxx
+
+
+    - name: Create Registration Assignment in resourcegroup level
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_registrationassignment.py]: + with randomly generating registration_assignment_id

File: plugins/modules/azure_rm_registrationassignment_info.py [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+    scope:
+        description:
+             - Scope of the registration assignment.
+        required: true
+        type: str
+    registration_assignment_id:
+        description:
+            - ID of the registration assignment.
+        type: str
+extends_documentation_fragment:
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_registrationassignment_info.py:33]: remove it.

File: plugins/modules/azure_rm_registrationdefinition.py [hunk 6]
```diff
@@ -0,0 +1,434 @@
...
+---
+module: azure_rm_registrationdefinition
+version_added: '1.3.0'
+short_description: Manage Azure RegistrationDefinition instance
+description:
+    - Create, update and delete instance of Azure RegistrationDefinition.
+options:
+    scope:
+        description:
+            - Scope of the resource.
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_registrationdefinition.py]: Scope of  registration definition.

File: plugins/modules/azure_rm_registrationdefinition.py [hunk 7]
```diff
@@ -0,0 +1,434 @@
...
+
+DOCUMENTATION = '''
+---
+module: azure_rm_registrationdefinition
+version_added: '1.3.0'
+short_description: Manage Azure RegistrationDefinition instance
+description:
+    - Create, update and delete instance of Azure RegistrationDefinition.
+options:
+    scope:
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_registrationdefinition.py]: does scope means the subscription in which the registration definition will be defined? If it is the case then it should be deleted, use the subscription provided by module parameter.

File: tests/integration/targets/azure_rm_registrationassignment/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,79 @@
+- name: set guid for registration assignment id
```
Comment (haiyuazhang) [inline tests/integration/targets/azure_rm_registrationassignment/tasks/main.yml]: name: set facts

File: tests/integration/targets/azure_rm_registrationdefinition/tasks/main.yml [hunk 9]
```diff
@@ -0,0 +1,108 @@
+- name: set guid for registration definition id
```
Comment (haiyuazhang) [inline tests/integration/targets/azure_rm_registrationdefinition/tasks/main.yml]: set facts

File: plugins/modules/azure_rm_registrationassignment.py (hunk 1 above)
Comment (Fred-sun) [reply plugins/modules/azure_rm_registrationassignment.py]: fixes

File: plugins/modules/azure_rm_registrationassignment.py (hunk 3 above)
Comment (Fred-sun) [reply plugins/modules/azure_rm_registrationassignment.py]: fixes

File: plugins/modules/azure_rm_registrationassignment.py (hunk 4 above)
Comment (Fred-sun) [reply plugins/modules/azure_rm_registrationassignment.py]: fixes

File: plugins/modules/azure_rm_registrationdefinition.py (hunk 6 above)
Comment (Fred-sun) [reply plugins/modules/azure_rm_registrationdefinition.py]: fixes

File: plugins/modules/azure_rm_registrationdefinition_info.py [hunk 10]
```diff
@@ -0,0 +1,241 @@
...
+
+DOCUMENTATION = '''
+---
+module: azure_rm_registrationdefinition_info
+version_added: '1.3.0'
+short_description: Get RegistrationDefinition info
+description:
+    - Get info of RegistrationDefinition.
+options:
+    scope:
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_registrationdefinition_info.py:24]: remove it!

File: plugins/modules/azure_rm_registrationdefinition.py (hunk 7 above)
Comment (Fred-sun) [reply plugins/modules/azure_rm_registrationdefinition.py]: I will delete it!

File: plugins/modules/azure_rm_registrationdefinition_info.py (hunk 10 above)
Comment (Fred-sun) [reply plugins/modules/azure_rm_registrationdefinition_info.py:24]: removed!

File: plugins/modules/azure_rm_registrationassignment_info.py (hunk 5 above)
Comment (haiyuazhang) [reply plugins/modules/azure_rm_registrationassignment_info.py:33]: my fault, sorry!

File: plugins/modules/azure_rm_registrationdefinition_info.py [hunk 11]
```diff
@@ -0,0 +1,241 @@
...
+    - azure.azcollection.azure_tags
+author:
+    - Fred-Sun (@Fred-Sun)
+'''
+
+EXAMPLES = '''
+    - name: Get Registration Definition
+      azure_rm_registrationdefinition_info:
+        registration_definition_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        scope: subscription/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_registrationdefinition_info.py]: scope should be deleted

Comment (Fred-sun) [reply plugins/modules/azure_rm_registrationdefinition_info.py]: already fixes

**Pull request merged**